src/reducers/select.js

```diff
@@ -27,6 +27,7 @@
 */
 import createDebugLogger from 'debug';
 import {normalizeSync} from 'normalize-diacritics';
+import clonedeep from 'lodash/clonedeep';
 
 export default (pattern) => (base, source) => {
```

Comment:
Make the arguments an object which support the equality function (Defaults to strict equality):
`export default ({pattern, equalityFunction = strictEquality}) =>`.

Also, the module needs to export the equality functions so that developer's can choose the built-in comparison functions:
```js
export function strictEquality() { ... }
export function subsetEquality() { ... }
```